buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTest.kt

```diff
@@ -63,6 +66,30 @@ open class DistributionTest : Test() {
         jvmArgumentProviders.add(libsRepository)
         systemProperty("java9Home", project.findProperty("java9Home") ?: System.getProperty("java9Home"))
     }
+
+    override fun executeTests() {
+        setTimeoutMonitor()
+        super.executeTests()
+    }
+
+    private
+    fun setTimeoutMonitor() {
+        Timer(true).schedule(timerTask {
+            project.javaexec {
+                classpath = project.getConfigurations().getByName("integTestRuntimeClasspath")
+                main = "org.gradle.integtests.fixtures.timeout.JavaProcessStackTracesMonitor"
+            }
+        }, determineTimeoutMillis())
+    }
+
+    private
+    fun determineTimeoutMillis(): Long {
+        return if ("embeded" == System.getProperty("org.gradle.integtest.executer")) {
+            TimeUnit.HOURS.toMillis(1) - TimeUnit.MINUTES.toMillis(1) // in case the build process killed by TC before it finishes
+        } else {
+            TimeUnit.HOURS.toMillis(3) - TimeUnit.MINUTES.toMillis(1)
```

Comment:
❌ Same here, I guess 1.5 hours should be fine: https://e.grdev.net/scans?list.offset=0&list.size=50&list.sortColumn=duration&list.sortOrder=desc&search.startTimeMax=1532436311443&search.startTimeMin=1531831511442&search.tags=BranchBuildAccept

Reply:
According to https://e.grdev.net/scans?list.offset=0&list.size=50&list.sortColumn=duration&list.sortOrder=desc&search.startTimeMax=1532436311443&search.startTimeMin=1531831511442 , lots of tests, especially PerformanceTest/AllCrossVersionTest, run over 2 hours. So I think 2-hour is a fair number.